assets/training/distillation/components/data_generation/src/common/constants.py

```diff
@@ -85,3 +85,23 @@ class InferenceMode:
     HFTV2_TEXT_GENERATION = "hftv2_text_generation"
     VLLM_CHAT_COMPLETION = "vllm_chat_completion"
     VLLM_TEXT_GENERATION = "vllm_text_generation"
+
+
+class MetaEnum(EnumMeta):
+    """Metaclass for Enum classes. to use the in operator to check if a value is in the Enum."""
+
+    def __contains__(cls, item):
+        """Check if the item is in the Enum."""
+        try:
+            cls(item)
+        except ValueError:
+            return False
+        return True
+
+
+class DataGenerationTaskType(str, Enum, metaclass=MetaEnum):
+    """Enum for data generation task types."""
+
+    NLI = "NLI"
```

Comment:
Could you please consider adding description/details for each of these so that it's easier for customer to make a choice? 

@kvijaykannan Could we please add these details in a verbose format in the documentation/blog?